code/game/objects/items/reagent_containers/glass.dm

```diff
@@ -539,6 +547,13 @@
 	to_chat(usr, SPAN_WARNING("[src] has no transfer control valve! Use a dispenser to fill it!"))
 	return
 
+/obj/item/reagent_container/glass/pressurized_canister/on_reagent_change()
+	update_icon()
+
+/obj/item/reagent_container/glass/pressurized_canister/update_icon()
```

Comment:
this should call parent for things like acid

Reply:
I looked at the other code in the file and both on_reagent_change and update_icon seem to work the same on things like bucket and so on?

Just to double-check, you want `on_reagent_change` to call `..()` always? I've looked at the `/reagent_containers` folder and most of the time `on_reagent_change` mostly just calls `update_icon()` minus food-related stuff that changes itself and `/obj/item/reagent_container/food/drinks/on_reagent_change()` that also updates its gulp size.

So I'm not entirely sure what would you like me to do exactly.

Reply:
I guess we could have a general function something for all item pouch things to update their icons on a child getting qdelled, but that would be a different PR I think. You'd have to register a signal listener whenever you put an item in and unregister it when you take thigns out, it would be a bit of work I'm guessing...

Reply:
Sorry, no, i'm talking about update_icon. The diff display in github is misleading, it shows the lines before.

Reply:
/obj/proc/update_icon handles effects overlays (in practice, acid)

Reply:
Updated the proc since this one doesn't have an issue. However I think a lot of other objects never call `/obj/proc/update_icon` since their parents `update_icon` would mess up their own icon. Like the bluespace beaker will never call beaker.update_icon since beaker has a fullness overlay bluespace beaker doesn't have...